scripts/upload_test_results/README.md

```diff
@@ -0,0 +1,40 @@
+# Auto-upload daily test results
+
+## Python script to automatically upload your HW Daily Test Results to the Github test_results repo 
+https://github.com/zephyrproject-rtos/test_results
+
+**To customize the script and run it on your machine, please modify code in Step 1 and Step 2 (Use Search to find that comments in the code of the script).**
+
+### Script description:
+1. The script creates a git directory, if a directory already created, script updates it (step 2-3) (-g argument)
+   * Fetches code from the source repo master branch zephyrproject-rtos/test_results
+   * Pushes updates to the origin repo master branch, in my case it is maksimmasalski/test_results
+2. Downloads versions.json file and finds the proper commit name for existing test results.
+   * When version found, it remembers its name and deletes versions.json file (-c argument)
+3. Creates a new branch, for example *intel_daily_master_177f2dac8b*
+4. Copies files from the Test results directory (-j argument) for a given commit to the .git repository *test_results*
+5. Renames all .xml files, for example from *junit-reel_board__zephyr_177f2dac8b.xml to reel_board.xml*
+6. The script does $git add for all test results files
+7. The script does $git commit
+8. The script does $git push 
+   * New branch will be created in your origin repo
+9. Checkouts back to the master branch. When you need to run the script again, all actions will be repeated again.
+Script execution finished.
+10. Now you manually open Github webpage and create the Pull Request to the upstream master branch. Github will generate a PR, you just click the "Compare & pull request" button on Github webpage.
+
+### Script usage:
+Please note that before you run script, you should enter your credentials to access Github in your Git.
+`python3 upload_to_github.py -c <commit> -j <test results path> -g <git dir path/test_results>`
+
+### Example:
+`python3 upload_to_github.py -c 177f2dac8b -j /home/ztest2/work/testreport/sumreport/master/master-177f2dac8b/junit4TestRail -g /home/maksim/test_results`
+
+### Script has 3 required arguments:
+`-c` Upload daily test results of that commit<br/>
+`-j` Directory with all daily test result files<br/>
+`-g` Directory to place git repo test_results<br/>
+
```

Comment:
It would be nice to have a -m argument to provide a commit message.

Reply:
I will add commit message and title into config.txt file

Reply:
> I will add commit message and title into config.txt file

I'm not sure config.txt would be a good fit.
Typically, in the commit message I'm dropping some test session abstract:
Since the test execution path is fully automated, this execution summary is the first point were I can get an idea on how things went (and also I guess this provides also confidence to the "merge authority").

So I need to be able to change the message at each execution (in an automated way, of course), editing config.txt doesn't seem a good place to do so.

Reply:
The console command will be too long if to add commit message there. Is it ok? I think it is not a problem...

Reply:
it is a problem. You can always provide default to the command line options, but using a file is really a bad alternative.
Not that you can still use a file to feed the options in, argparse does support this.

Reply:
So add -m for message?

Reply:
I added -t for title, and -m for message